internal/controllers/provisioningrequest_controller.go

```diff
@@ -280,6 +289,31 @@ func (t *provisioningRequestReconcilerTask) run(ctx context.Context) (ctrl.Resul
 
 	}
 
+	// Check if NodePool configured status is set after the cluster provisioning status is fulfilled
```

Comment:
I just realized since upgrade is supported now(line 277-286), we also need to take that into account. It seems like checkNodePoolConfigStatus should be moved above which prevent further updates if nodePool config is still in progress? 🤔

Reply:
It has been moved up.